### PR TITLE
[needs-docs] Make maptip and geometry attribute optional in featureinfo response

### PR DESCRIPTION
**Pull request by mhugent**
In the QGIS Server GetFeatureInfo response, there is the maptip attribute and the geometry attribute (if enabled in the project). This pull request changes the behaviour to only include these attributes if requested by the client (new parameters WITH_GEOMETRY and WITH_MAPTIP).  Some wms clients always display popups and they can get quite large if the geometry wkt is contained in the response.

### CODE REVIEW

File: src/server/services/wms/qgswmsparameters.h [hunk 1]
```diff
@@ -916,6 +918,18 @@ namespace QgsWms
        */
       QString externalWMSUri( const QString &id ) const;
 
+      /**
+       * @brief Returns if the client wants the feature info response with geometry information
+       * @return true if geometry information is requested for feature info response
```
Comment (pblottiere) [inline src/server/services/wms/qgswmsparameters.h]: It should be `\brief` and `\returns`

File: src/server/services/wms/qgswmsparameters.h [hunk 2]
```diff
@@ -916,6 +918,18 @@ namespace QgsWms
...
+       */
+      bool withGeometry() const;
+
+      /**
+       * @brief withMapTip
+       * @return true if maptip information is requested for feature info response
+       */
```
Comment (pblottiere) [inline src/server/services/wms/qgswmsparameters.h]: `\brief` and `\returns`

**Comment by pblottiere** [comment]
Thank you @mhugent! I think it's a good move to have a light `GetFeatureInfo` by default :+1:!

**Comment by mhugent** [comment]
Thanks for reviewing, @pblottiere !

**Comment by pblottiere** [comment]
By the way, I think that the tag for documentation is `[needs-docs]` instead of `[needs-documentation]`. But I don't know if it's important. I mean, is it parsed somewhere to create a proper corresponding issue in QGIS-Documentation? @m-kuhn?

**Comment by m-kuhn** [comment]
As far as I know the `[needs-docs]` tag gets parsed in some github hook and opens an issue in the documentation repository.

**Comment by Gustry** [comment]
@pblottiere https://github.com/qgis/QGIS-Documentation/issues?q=is%3Aissue+is%3Aopen+label%3A%22Automatic+new+feature%22

**Comment by pblottiere** [comment]
> As far as I know the [needs-docs] tag gets parsed in some github hook and opens an issue in the documentation repository.

OK so @mhugent, you should update the title of your PR :)

**Review by elpaso — APPROVED** [review]
Good move, thanks!

**Comment by pblottiere** [comment]
@mhugent Before merging this PR, can you add a test for WITH_MAPTIP?

**Review by rldhont — APPROVED** [review]
We will need to update Lizmap! Thanks

**Comment by andreasneumann** [comment]
@rldhont - yes sorry about that. QWC2 also needs updates. But I still make this makes sense to get rid of geometry and maptip by default - as these do not make much sense in a Desktop-GIS.

It is rather easy to change this in Lizmap and QWC2 and a one-time effort.

Thanks for your understanding!

**Comment by rldhont** [comment]
@andreasneumann lizmap already parses and transforms the GetFeatureInfo response so we just have to update it to use this params.

**Pull request merged**